File: backend/store/app/order/query/order.go
package query

import (
	"store/app/domain/data"
	"time"
)

type Order struct {
	Id        string
	Number    string
	CreatedAt time.Time
	Customer  *Customer
	Status    string
	Items     []*OrderItem
}

type OrderItem struct {
	Book *Book
	Qty  int
}

type Customer struct {
	Name            string
	Phone           string
	DeliveryAddress string
}

func (Order) fromDataObject(order *data.Order) *Order {
	items := []*OrderItem{}
	for _, dataItem := range order.Items {
		items = append(items, OrderItem{}.fromDataObject(dataItem))
	}

	return &Order{
		Id:        order.Id,
		Number:    order.Number,
		CreatedAt: order.CreatedAt,
		Customer:  Customer{}.fromDataObject(order.Customer),
		Status:    string(order.Status),
		Items:     items,
	}
}

func (OrderItem) fromDataObject(item *data.OrderItem) *OrderItem {
	return &OrderItem{
		Book: Book{}.fromDataObject(item.Book),
		Qty:  item.Qty,
	}
}

func (Customer) fromDataObject(customer *data.Customer) *Customer {
	return &Customer{
		Name:            customer.Name,
		Phone:           customer.Phone,
		DeliveryAddress: customer.DeliveryAddress,
	}
}

File: backend/ecommerce/src/main/java/com/bookstore/ecommerce/app/order/query/dto/Order.java
package com.bookstore.ecommerce.app.order.query.dto;

import java.time.Instant;
import java.util.List;
import java.util.stream.Collectors;
import lombok.Builder;
import lombok.Data;

@Data
@Builder
public class Order {
  private String id;
  private String number;
  private String status;
  private Instant createdAt;
  private Customer customer;
  private List<OrderItem> items;

  public static Order fromDataObject(com.bookstore.ecommerce.app.domain.data.Order order) {
    return Order.builder()
      .id(order.getId())
      .number(order.getNumber())
      .status(order.getStatus())
      .createdAt(order.getCreatedAt())
      .customer(Customer.builder()
        .id(order.getCustomerId())
        .name(order.getCustomerName())
        .phone(order.getCustomerPhone())
        .deliveryAddress(order.getCustomerDeliveryAddress())
        .build())
      .items(order.getItems().stream()
        .map(item -> OrderItem.fromDataObject(item))
        .collect(Collectors.toList()))
      .build();
  }
}
